src/detect-detection-filter.c

```diff
@@ -554,81 +518,71 @@ static int DetectDetectionFilterTestSig3(void)
 
     DetectEngineCtx *de_ctx = DetectEngineCtxInit();
-    if (de_ctx == NULL) {
-        goto end;
-    }
+    FAIL_IF_NULL(de_ctx);
 
     de_ctx->flags |= DE_QUIET;
 
-    s = de_ctx->sig_list = SigInit(de_ctx,"drop tcp any any -> any 80 (msg:\"detection_filter Test 2\"; detection_filter: track by_dst, count 2, seconds 60; sid:10;)");
-    if (s == NULL) {
-        goto end;
-    }
+    Signature *s = DetectEngineAppendSig(de_ctx,
+            "drop tcp any any -> any 80 (msg:\"detection_filter Test 2\"; "
+            "detection_filter: track by_dst, count 2, seconds 60; sid:10;)");
+    FAIL_IF_NULL(s);
 
     SigGroupBuild(de_ctx);
     DetectEngineThreadCtxInit(&th_v, (void *)de_ctx, (void *)&det_ctx);
 
     TimeGet(&p->ts);
 
     SigMatchSignatures(&th_v, de_ctx, det_ctx, p);
-    alerts = PacketAlertCheck(p, 10);
-    drops += ((PacketTestAction(p, ACTION_DROP)) ? 1 : 0);
+    FAIL_IF(PacketAlertCheck(p, 10));
+    FAIL_IF(PacketTestAction(p, ACTION_DROP));
     p->action = 0;
 
     SigMatchSignatures(&th_v, de_ctx, det_ctx, p);
-    alerts += PacketAlertCheck(p, 10);
-    drops += ((PacketTestAction(p, ACTION_DROP)) ? 1 : 0);
+    FAIL_IF(PacketAlertCheck(p, 10));
+    FAIL_IF(PacketTestAction(p, ACTION_DROP));
     p->action = 0;
 
     SigMatchSignatures(&th_v, de_ctx, det_ctx, p);
-    alerts += PacketAlertCheck(p, 10);
-    drops += ((PacketTestAction(p, ACTION_DROP)) ? 1 : 0);
+    FAIL_IF_NOT(PacketAlertCheck(p, 10));
+    FAIL_IF_NOT(PacketTestAction(p, ACTION_DROP));
     p->action = 0;
 
     TimeSetIncrementTime(200);
     TimeGet(&p->ts);
 
     SigMatchSignatures(&th_v, de_ctx, det_ctx, p);
-    alerts += PacketAlertCheck(p, 10);
-    drops += ((PacketTestAction(p, ACTION_DROP)) ? 1 : 0);
+    FAIL_IF(PacketAlertCheck(p, 10));
+    FAIL_IF(PacketTestAction(p, ACTION_DROP));
     p->action = 0;
 
     SigMatchSignatures(&th_v, de_ctx, det_ctx, p);
-    alerts += PacketAlertCheck(p, 10);
-    drops += ((PacketTestAction(p, ACTION_DROP)) ? 1 : 0);
+    FAIL_IF(PacketAlertCheck(p, 10));
+    FAIL_IF(PacketTestAction(p, ACTION_DROP));
     p->action = 0;
 
     SigMatchSignatures(&th_v, de_ctx, det_ctx, p);
-    alerts += PacketAlertCheck(p, 10);
-    drops += ((PacketTestAction(p, ACTION_DROP)) ? 1 : 0);
+    FAIL_IF_NOT(PacketAlertCheck(p, 10));
+    FAIL_IF_NOT(PacketTestAction(p, ACTION_DROP));
     p->action = 0;
 
     SigMatchSignatures(&th_v, de_ctx, det_ctx, p);
-    alerts += PacketAlertCheck(p, 10);
-    drops += ((PacketTestAction(p, ACTION_DROP)) ? 1 : 0);
+    FAIL_IF_NOT(PacketAlertCheck(p, 10));
+    FAIL_IF_NOT(PacketTestAction(p, ACTION_DROP));
     p->action = 0;
 
-    if (alerts == 3 && drops == 3)
-        result = 1;
-    else {
-        if (alerts != 3)
-            printf("alerts: %d != 3: ", alerts);
-        if (drops != 3)
-            printf("drops: %d != 3: ", drops);
-    }
-
     SigGroupCleanup(de_ctx);
     SigCleanSignatures(de_ctx);
```

Comment:
These can be removed from this test and the others that have used `DetectEngineAppendSig`